lib/internal/http2/core.js

```diff
@@ -1844,7 +1871,31 @@ function createServer(options, handler) {
 // HTTP2-Settings header frame.
 function getPackedSettings(settings) {
   assertIsObject(settings, 'settings');
-  return binding.packSettings(settings || Object.create(null));
+  settings = Object.assign(Object.create(null), settings);
+  assertWithinRange('headerTableSize',
+                    settings.headerTableSize,
+                    0, 2 ** 32 - 1);
+  assertWithinRange('initialWindowSize',
+                    settings.initialWindowSize,
+                    0, 2 ** 32 - 1);
+  assertWithinRange('maxFrameSize',
+                    settings.maxFrameSize,
+                    16384, 2 ** 24 - 1);
+  assertWithinRange('maxConcurrentStreams',
+                    settings.maxConcurrentStreams,
+                    0, 2 ** 31 - 1);
+  assertWithinRange('maxHeaderListSize',
+                    settings.maxHeaderListSize,
+                    0, 2 ** 32 - 1);
+  if (settings.enablePush !== undefined &&
+      typeof settings.enablePush !== 'boolean') {
```

Comment:
I like how this is written but wonder if it is more efficient to do a strict comparison.

```js
if (settings.enablePush !== undefined &&
    settings.enablePush !== true &&
    settings.enablePush !== false) {
```

Cf [lodash.isBoolean()](https://github.com/lodash/lodash/blob/4.17.4/lodash.js#L11396)